### ISSUE TITLE
Backfill stuck on `Failed to send scheduled backfill work`

### ISSUE THREAD

**Issue by realbigsean** [sigp/lighthouse#5504]
## Description

User has reported stuck backfill which happens after this log:

```
Mar 29 07:08:06.001 ERRO Failed to send scheduled backfill work, info: sending work back to queue, service: bproc, module: beacon_processor::work_reprocessing_queue:884
```
We do re-insert the batch into `queued_backfill_batches` [here](https://github.com/sigp/lighthouse/blob/stable/beacon_node/beacon_processor/src/work_reprocessing_queue.rs#L889). Do we need to also update `next_backfill_batch_event` like we do [here](https://github.com/sigp/lighthouse/blob/stable/beacon_node/beacon_processor/src/work_reprocessing_queue.rs#L751-L753)?

**Comment by jimmygchen** [comment]
Oh yeah nice catch!  

I think the scenario is:
1. There is 1 queued backfill batch in the reprocessing queue
2. At the scheduled interval, it's pop'd from the queue for processing. At this point, it [recomputes the `next_backfill_batch_event`](beacon_node/beacon_processor/src/work_reprocessing_queue.rs:338) when we remove a queued batch. This always set the `next_backfill_batch_event` to `None` as there's nothing to schedule after we pop the item from the queue. (LH currently process 1 at a time)
3. However the receiver queue for `ReadyWork` is full, and it fails to send, so we insert it back to the queue (without scheduling a batch event like @realbigsean suggested)
4. Backfill is stuck now because we don't trigger another batch until the queued one is processed.

**Comment by jimmygchen** [comment]
Resolved in #5575
